### PR TITLE
New NPM package: handlebars-i18next

### PR DESCRIPTION
**Pull request by jgonggrijp**
This is a reusable piece of glue code between Handlebars and i18next, which adds a very nice i18n syntax to templates (see README). Like jquery-promise, I extracted this from readit-interface.

There is a reason for forcing users to pass `Handlebars` and an `i18next` instance to the exported function. I now realise I didn't document that reason yet. I can still add it.

### CODE REVIEW

**Review by alexhebing — APPROVED** [review]
It looks good! Can I try it out in READ-IT soon?

**Comment by jgonggrijp** [comment]
Yes you can! Although we have no features yet... Let’s chat about it next week.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,171 @@
...
+template({name: 'Alice'});
+```
+
+Result
+
+```
+Hello, Alice!
+```
+
+Properties in the context of the helper are automatically available as interpolation values to i18next. It just works!
```
Comment (oktaal) [inline README.md:61]: I'm not quite sure what is meant by "context of the helper". E.g. a `name` property on `this`?

File: handlebars-i18next.es6 [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        target[key] = source[key];
+    }
+    return target;
+}
+
+export default function(Handlebars, i18next, name = 'i18n') {
+    Handlebars.registerHelper(name, function(key, {hash, data, fn}) {
+        let final = {returnObjects: false};
+        for (let key of jsonKeys) if (hash[key]) {
+            final[key] = JSON.parse(hash[key]);
```
Comment (oktaal) [inline handlebars-i18next.es6]: It could use some explanation about why these fields need this treatment.

File: handlebars-i18next.es6 [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const jsonKeys = ['lngs', 'fallbackLng', 'ns', 'postProcess', 'interpolation'];
+
+function extend(target, ...sources) {
```
Comment (oktaal) [inline handlebars-i18next.es6]: Maybe state why `Object.assign` hasn't been used here instead. Compatibility with older browsers?

**Review by oktaal — COMMENTED** [review]
Just some minor uncertainty here.

File: README.md (hunk 1 above)
Comment (jgonggrijp) [reply README.md:61]: This is Handlebars jargon. If you have mustaches in a template, like this,

```hbs
{{#mouse}}
<h1>{{whiskers}}</h1>
<p>{{fur}}</p>
{{/mouse}}
```

and you call the compiled template with a *root context*, like this,

```js
const rootContext = {
    mouse: {
        whiskers: 'Curly whiskers',
        fur: '... lots of fuzzy text ...',
    },
};

let result = compiledTemplate(rootContext);
```

then `rootContext.mouse` is the context of `{{whiskers}}` and `{{fur}}`. I expect frequent Handlebars users to know this from [the documentation](http://handlebarsjs.com) as well as the [Mustache docs](http://mustache.github.io/mustache.5.html).

The context in which a helper operates happens to be passed as `this` to the function in which the helper is implemented, but end users should not need to worry about this.

File: handlebars-i18next.es6 (hunk 2 above)
Comment (jgonggrijp) [reply handlebars-i18next.es6]: Will add!

File: handlebars-i18next.es6 (hunk 3 above)
Comment (jgonggrijp) [reply handlebars-i18next.es6]: Exactly (IE11). Will add.

Or I could create a `lodash-object-assign` polyfill package and thell people to use that if they need it. But that's probably stretching it.

Comment (oktaal) [reply handlebars-i18next.es6]: I agree. If this were a bigger library I'd just expect people to be using some form of polyfill.

**Pull request merged**